src/monitor.rs

```diff
@@ -141,12 +144,39 @@ macro_rules! monitor_cmd_for {
 					}
 
 					let (solution, score) =
-					match crate::helpers::[<mine_solution_$runtime>](&api, Some(hash), config.solver).timed().await {
+					match helpers::[<mine_solution_$runtime>](&api, Some(hash), config.solver).timed().await {
 						(Ok((solution, score, size)), elapsed) => {
 							let elapsed_ms = elapsed.as_millis();
-							crate::prometheus::observe_mined_solution_duration(elapsed_ms as f64);
-							crate::prometheus::set_score(score);
-							log::trace!(target: LOG_TARGET, "Mined solution with {:?} size: {:?} round: {:?} at: {}, took: {} ms", score, size, round, at.number(), elapsed_ms);
+							let encoded_len = solution.encoded_size();
+							let active_voters = solution.voter_count() as u32;
+							let desired_targets = solution.unique_targets().len() as u32;
+
+							let final_weight = tokio::task::spawn_blocking(move || {
```

Comment:
@kianenigma this will spawn the on task on the blocking thread pool in the tokio

if we call it on the `async scheduler` it will fuck up the scheduler i.e, in `tokio::task` so this should work now.

Reply:
it basically hack around that we are forced to block the current thread because of `MinerConfig::solution_weight`